packages/engine/lib/memory/src/error.rs

```diff
@@ -84,6 +84,22 @@ pub enum Error {
     ColumnNotFound(String),
 }
 
+impl From<shared_memory::ShmemError> for Error {
+    fn from(e: shared_memory::ShmemError) -> Self {
+        if cfg!(target_os = "macos") {
+            if matches!(e, shared_memory::ShmemError::UnknownOsError(22)) {
+                tracing::error!(
```

Comment:
This is an interesting pattern, I hadn't thought about using an impl of the error to intercept and log...

I'm fine with us implementing it because it does have immediate value but I don't think this is really how we'd want to do things, especially when we move to `error-stack`. I think we'd want this `attach`ed as a 'Suggestion' or something which our error handling strategy prints.

Reply:
Worth capturing a `@todo`?

Reply:
I don't think so :) This entire module will be invalidated as we migrate to `error-stack`. Basically anywhere we create or use errors in the Engine will be updated so we'd see this and change it.

Reply:
Attaching those things to `error-stack` will be much easier with #794, as you can provide a hook for types, so it's not required anymore to implement `Display` and call `attach_printable` but you can instead call `install_debug_hook::<Suggestion>(|suggestion, _| ...)`, which is used for printing (instead of "opaque attachment").